language/move-binary-format/src/file_format.rs

```diff
@@ -1994,7 +1976,8 @@ pub fn dummy_procedure_module(code: Vec<Bytecode>) -> CompiledModule {
 
     module.function_handles.push(fun_handle);
     module.function_defs.push(fun_def);
-    module.freeze().unwrap()
+    BoundsChecker::verify_module(&module).expect("invalid bounds in module");
```

Comment:
What is this function still used for? seems weird and bad

Reply:
This function is only used by bytecode-verifier-tests, so #8735 might be an improvement: it moves the function, which seems like a convenience helper for tests, out of this core module and closer to the tests that use it, preventing the function from being used more widely.